apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-plugin/src/main/java/co/elastic/apm/agent/opentelemetry/sdk/BehavioralAttributes.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.opentelemetry.sdk;
+
+/**
+ * An extension to the OTel API that allows us to provide special tracing settings.
+ * Such settings may affect tracing behavior internally, but they are not added as data and not persisted as span attributes.
+ */
+public class BehavioralAttributes {
+    public static final String NON_DISCARDABLE = "Elastic-OpenTelemetry-NonDiscardable";
```

Comment:
[minor] OTel attribute keys are lower-cased and separated by dots, I would suggest to change the value to something like `elastic.otel.nondiscardable`
```suggestion
    public static final String NON_DISCARDABLE = "elastic.otel.nondiscardable";
```

Also, to prevent allocation it is recommended to store the attribute as an `AttributeKey` rather than use the `String` key.

Reply:
Good point! I started implementation from the builder where preventing this allocation didn't make sense, but it does in this API